src/ecpair.js

```diff
@@ -57,46 +58,25 @@ ECPair.fromPublicKeyBuffer = function (buffer, network) {
   })
 }
 
-ECPair.fromWIF = function (string, networks) {
-  var payload = bs58check.decode(string)
-  var version = payload.readUInt8(0)
-  var compressed
+ECPair.fromWIF = function (string, network) {
+  network = network || NETWORKS.bitcoin
+  var buffer = bs58check.decode(string)
 
-  if (payload.length === 34) {
-    if (payload[33] !== 0x01) throw new Error('Invalid compression flag')
+  if (types.Array(network)) {
+    var version = buffer[0]
 
-    // truncate the version/compression bytes
-    payload = payload.slice(1, -1)
-    compressed = true
-
-  // no compression flag
-  } else {
-    if (payload.length !== 33) throw new Error('Invalid WIF payload length')
-
-    // Truncate the version byte
-    payload = payload.slice(1)
-    compressed = false
-  }
-
-  var network
-
-  // list of networks?
-  if (Array.isArray(networks)) {
-    network = networks.filter(function (network) {
+    network = network.filter(function (network) {
       return version === network.wif
     }).pop() || {}
 
-  // otherwise, assume a network object (or default to bitcoin)
-  } else {
-    network = networks || NETWORKS.bitcoin
+    if (version !== network.wif) throw new Error('Invalid network version')
   }
 
-  if (version !== network.wif) throw new Error('Invalid network')
-
-  var d = BigInteger.fromBuffer(payload)
+  var decoded = wif.decodeRaw(network.wif, buffer)
```

Comment:
Can we please not swap the order of version and buffer here? I'd prefer it being consistent for bitcoinjs-lib and wif.